tests/Collection/CrudSpecFunctionalTest.php

```diff
@@ -116,10 +128,9 @@ private function executeOperation(array $operation)
                 );
 
             case 'bulkWrite':
-                $results = $this->prepareBulkWriteArguments($operation['arguments']);
                 return $this->collection->bulkWrite(
-                    array_diff_key($results, ['options' => 1]),
-                    $results['options']
+                    array_map([$this, 'prepareBulkWriteRequest'], $operation['arguments']['requests']),
```

Comment:
The original `prepareBulkWriteArguments()` method returned an array consisting of both requests (under numeric indices) and options (under a single associative index). I refactored that method to instead format each request element independently (hence the use of `array_map`).